control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -2688,6 +2692,19 @@ func (r *HostedControlPlaneReconciler) reconcileManagedEtcd(ctx context.Context,
 		}
 	}
 
+	// Block to recover the ETCD cluster if it is not ready after etcdRecoveryPeriod
+	timePassed := time.Since(statefulSet.ObjectMeta.CreationTimestamp.Time)
+	ready := util.IsStatefulSetReady(ctx, statefulSet)
+	// If the etcd cluster is not ready after etcdRecoveryPeriod, we will attempt to recover it
+	// but we will stop trying it after etcdRecoveryTimeout to avoid loss of data in other scenarios.
+	if !ready && (timePassed > etcdRecoveryPeriod && timePassed <= etcdRecoveryTimeout) {
```

Comment:
@jparrill Is this possible:
- We have etcd quorum and HC is happy
- statefulSet goes not ready because one single replica failure. E.g. recycling management data plane
- The HC has live for less than etcdRecoveryTimeout
- We destroy etcd for a legit working cluster

I think this should be gated on the HC ever being available for the first time

Reply:
Following up here https://github.com/openshift/hypershift/pull/4419